moped-editor/src/views/projects/projectView/ProjectComponents/utils/useToolbarErrorMessage.js

```diff
@@ -0,0 +1,29 @@
+import { useReducer } from "react";
```

Comment:
TIL `useReducer`. Thanks for that. I think this is a great managing complex state without getting something like `redux` involved. Super nice.